contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -576,6 +603,14 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
                     : sqrtRatioX96TickHigher
             })
         );
+
+        /**
+         * In the interest of each function in _rebalance to leave the contract state as
+         * clean as possible the OETHb tokens here are burned. This decreases the
+         * dependence where `_swapToDesiredPosition` function relies on later functions
+         * (`addLiquidity`) to burn the OETHb. Reducing the risk of error introduction.
+         */
+        _burnOethbOnTheContract();
```

Comment:
⚪️ So currently, we do this to ensure that there's no leftover SuperOETHb on the contract.

<img width="1200" alt="image" src="https://github.com/user-attachments/assets/277a02dd-475b-41c9-9979-0d20a4cfa3ff">

This is ugly because two of the inner methods that can get superOETHb, check and burn locally. However the remove liquidity side is currently burning at different higher levels, two of which are in external methods, one of which is internal.

We could make this very simple by burning, and only burning, right at the places we might get superOETHb:

<img width="1224" alt="image" src="https://github.com/user-attachments/assets/12925b82-4e61-4b06-a4b7-36ce0fb686e8">

Reply:
After some discussion with shah in discord, the current contract burns up to three times on a deposit or rebalance.

If we moved the burnOETHb calls to the following locations, we could exactly one possible burn per external call:

<img width="783" alt="image" src="https://github.com/user-attachments/assets/2f0d12a0-4d3d-4b49-85b9-66096b376105">

Reply:
Our primary objective here is to have the burning happen at one consistent level of the codebase. This makes the code understandable and hard to have bugs, because there's a simple rule, any where it's not followed is probably a bug.

So either burn only at the lowest level. (simple, hard to mess up, runs more often than needed on trade rebalances, but only once on everything else). Or run it at a higher level in the code only, and we ensure that it only runs once.

Here's current state.

<img width="950" alt="image" src="https://github.com/user-attachments/assets/4b854a2c-2b1c-478a-b74c-a8f4962e94ae">

What is a nightmare is running it at many different levels. This requires more calls (called five times vs three), is much harder to understand, and much easier to make a mistake on in the future. There's no underlying rule about where the burn code is that people can use to think about the code, or fit new code in. The current code not only keeps it called at multiple levels of the code, it doesn't simplify the number calls burns to burn, and also does the extra dust burns.

We need to either run the burns only at the low level in `_removeLiquidity`, `_swapToDesiredPosition`, and `_addLiquidity`, and accept an extra dust burn on rebalances+trades, or run it only in `withdrawAll`, `withdraw`, and in `_rebalance` (after the call to ` _addLiquidity` in `_rebalance`).

Reply:
I've created a separate PR for this. https://github.com/OriginProtocol/origin-dollar/pull/2290

I think it make sense to improve the accuracy of removing liquidity and add thresholds to triggers adding liquidity so the tests capture any potential issues with lingering OETHb.

Reply:
Just read the edited comment, yes I agree completely those are the 2 sensible options. In the PR linked in the comment above is also a change where OETHb is burned only at the lowest level

Reply:
Alright. Now we are cooking. 
<img width="879" alt="image" src="https://github.com/user-attachments/assets/12cd23f6-2316-457a-b4e9-566529798714">